### PR TITLE
Add a notice when tags and attributes are rejected

### LINKED ISSUE #1140: Inform the user of disallowed HTML
There have been a lot of question on Discord and Meta about posts being different from what was shown in the preview. Currently, due to the way we handle submissions, differences between the post preview and the actual post comes from sanitization.

Could we have something that automatically detects when the sanitizer would remove content, and inform the user? We could do this either client-side or server side.

On the client side, we could have a script to detect when the preview contains disallowed tags and attributes, in accordance with https://meta.codidact.com/posts/277420. This would give live feedback, but could suffer from any discrepancies between the "client sanitizer" and the true sanitizer.

On the server side, we check that the submitted HTML is the same before and after sanitization, and if different, we would inform the users similar to how we inform the user when they are submitting an image without alt text. This would be accurate, but only catchable when the user hits the submit button.

### PR DESCRIPTION
**Pull request by MoshiKoi**
Resolves #1140 

image

Currently waiting for a help article to be created so I can link to that instead of my own post.

### CODE REVIEW

**Comment by cellio** [comment]
Is someone working on that help article?  If not, what do we need -- an edit to the formatting topic, a new topic, something else?

**Comment by MoshiKoi** [comment]
I don't know if anyone is working on it; pinging @codidact/documentation

**Comment by MoshiKoi** [comment]
I'm not sure what exactly is needed; I think any of your suggestions would work fine. I'll leave it to the writers to figure it out 🙂

**Comment by superplane39** [comment]
Help article written; it currently exists on Meta and #1212 adds it to seeds for the network

File: app/assets/javascripts/posts.js
Comment (Oaphi) [inline app/assets/javascripts/posts.js:21]: I've added a hook to also sanitize values of the `rowspan` and `colspan` attributes to disallow non-numeric values (might be worth to allow only ints too) - since we have such a powerful tool now, why not?

File: app/assets/javascripts/posts.js
Comment (Oaphi) [inline app/assets/javascripts/posts.js:179]: Fixed a small issue with the notice not displaying when there are no removed elements, but some of the attributes are.

**Review by Oaphi — APPROVED** [review]
I've added a small fix & a minor expansion for the PR (someone else might want to review them), so other than that - LGTM! Played around with the changes locally, seems to be working perfectly fine.

**Comment by cellio** [comment]
@MoshiKoi  can you review the latest changes and let us know if you're happy with them?

**Comment by MoshiKoi** [comment]
Looks good to me

**Pull request merged**
